doc/design.md

```diff
@@ -274,3 +274,34 @@ Notes:
 
 - Store passwords encrypted
 - Use Postgres features for exporting JSON
+
+Database Functions
+--------------
+Functions that query or update the database will be organized into a single module in the expungeservice.
+
+    create_user(database, email, admin, hashed_password)
+
+Insert a new user into the Users table with the given email string and admin flag, and inserts the password hash into the Auth table and link it with the User uuid. The uuid of each, and the date_created and date_updated are generated within the database.
+
+This pair of inserts is an atomic operation, so if one fails then the other one is cancelled and has no effect.
+Returns None if insert is successful; otherwise throws an informative error.
```

Comment:
From working with other ORM's I'm used to seeing it return an object, in this case a user object (or True), if it succeeds; otherwise an error (or a falsey value). That way the endpoint code would be something like this:
```
if create_user(database, email, admin, hashed_password):
    process_request
else:
     return to user creation screen with failure notification
```
or do you have something else in mind?

Reply:
@NickSchimek Yes, returning True makes intuitive sense, to make your use case syntax work. Though when I thought through possible sources of failure, it seemed like it never makes sense to return False; it should just throw the specific error (Since an error object isn't falsey). Making the "True" return value technically spurious. But I think it works equally well so there's no reason not to change this to returning True.

For returning a User object: again this would also be fine I think, but the caller already has the email, password_hash, and created/modified time values; the only thing they don't have is the UUID which I don't know is useful in an endpoint?

Reply:
@wittejm Agreed!, I think it's not so much as what the caller knows already as to the flexibility of the ORM. However I'm fine either way: True/False, True/error, Object/False, Object/error. <- there are use cases for each one of the scenarios, but I would advocate to only build what we need while allowing it to be open for extensions in the future.